docs/source/manual/snakemake.md

```diff
@@ -1,29 +1,97 @@
+
+To construct a graphical interface from a snakemake workflow, the parameters need to be explicitly identified and defined so that they can be presented to scientist to be filled out through a web application. The `latch_metadata.py` file holds these parameter definitions, along with any styling or cosmetic modifications the developer wishes to make to each parameter.
+
+Asking the question, "what are the minimum set of files needed to run my workflow to completion" is a good way to begin identifying what these parameters might be. 
+
+An example of this `latch_metadata.py` file:
+
+```
+# latch_metadata.py
+
+from pathlib import Path
+
+from latch.types.directory import LatchDir
+from latch.types.file import LatchFile
+from latch.types.metadata import LatchAuthor, SnakemakeFileParameter, SnakemakeMetadata
+
+SnakemakeMetadata(
+    display_name="fgbio Best Practise FASTQ -> Consensus Pipeline",
+    author=LatchAuthor(
+        name="Fulcrum Genomics",
+    ),
+    parameters={
+        "r1_fastq": SnakemakeFileParameter(
+            display_name="Read 1 FastQ",
+            type=LatchFile,
+            path=Path("r1.fq.gz"),
+        ),
+        "r2_fastq": SnakemakeFileParameter(
+            display_name="Read 2 FastQ",
+            type=LatchFile,
+            path=Path("r2.fq.gz"),
+        ),
+        "genome": SnakemakeFileParameter(
+            display_name="Reference Genome",
+            type=LatchDir,
+            path=Path("hs38DH"),
+        ),
+    },
+)
+```
+
+### Defining all dependencies in a container
+
+When snakemake jobs are executed on Latch, they will run in an environment defined by a `Dockerfile`. Any dependencies, such as third-party binaries, python libraries, shell scripts, need to be installed and configured correctly in this `Dockerfile` so the job has access to them.
+
+To generate a `Dockerfile` that can be modified, the following command is available:
+
+`latch dockerfile <workflow_folder>`
+
+A container is automatically built from the `Dockerfile` generated by this command when your snakemake project is registered with Latch.
+
+### Ensure your `Snakefile` is compatible with cloud execution
+
+When snakemake workflows are executed on Latch, each generated job is run in a separate container on a potentially isolated machine. This means your `Snakefile` might need to be modified to address problems that arise from this type of execution that were not present when executing locally:
 
-- Add basic workflow metadata: name and author
-- Find input files and specify them as workflow parameters
-  - If necessary, include a `config.yaml` file in the workflow image that matches the parameter specification
-- Add missing runtime dependencies (conda, pip, system packages)
 - Add missing rule inputs that are implicitly fulfiled when executing locally. Index files for biological data are commonly expected to always be alongside their matching data.
```

Comment:
We should have link outs to specific examples here too (maybe add examples in troubleshooting and link to those) 

I am reading this doc to figure out what actions I should take, but it's really hard to do that without examples